File: 3396-minimum-number-of-operations-to-make-elements-in-array-distinct/3396-minimum-number-of-operations-to-make-elements-in-array-distinct.cpp
class Solution {
public:
    int minimumOperations(vector<int>& nums) {
        reverse(nums.begin(),nums.end());
        int cnt[105];
        memset(cnt,0,sizeof(cnt));
        for(auto it:nums)cnt[it]++;
        for(int i = 0;;i++){
            int ok=1;
            for(int j = 1;j<=100;j++){
                if(cnt[j]>1)ok=0;
            }
            if(ok)return i;
            for(int j =0;j<3;j++){
                cnt[nums.back()]--;
                nums.pop_back();
                if(nums.empty())break;
            }
        }
        return 0;
    }
};

File: 3396-minimum-number-of-operations-to-make-elements-in-array-distinct/3396-minimum-number-of-operations-to-make-elements-in-array-distinct.java

class Solution {

    boolean ok(int[] a)
    {
        Set<Integer> s = new HashSet<>();
        for(int v : a){
            if(!s.add(v))return false;
        }
        return true;
    }
    public int minimumOperations(int[] nums) {
        int step = 0;
        while(true){
            if(ok(nums)){
                return step;
            }
            step++;
            if(nums.length <= 2){
                nums = new int[0];
            }else{
                nums = Arrays.copyOfRange(nums, 3, nums.length);
            }
        }
    }
}